src/TinyIoC/TinyIoC.cs

```diff
@@ -3619,19 +3619,10 @@ private bool IsIgnoredAssembly(Assembly assembly)
 
         private bool IsIgnoredType(Type type, Func<Type, bool> registrationPredicate)
         {
-            if (registrationPredicate != null && !registrationPredicate(type))
-            {
-                ignoreChecks.Add(t => !registrationPredicate(t));
+            if (ignoreChecks.Any(c => c(type)))
                 return true;
-            }
 
-            for (int i = 0; i < ignoreChecks.Count; i++)
-            {
-                if (ignoreChecks[i](type))
-                    return true;
-            }
-
-            return false;
+            return registrationPredicate != null && !registrationPredicate(type);

```

Comment:
Do we know the context on why it was checking the registration Predicate first? Can you please request a review from the person who wrote this?

Reply:
I think the intention of @AlexanderMy was to early return if the predicate matches but then he forgot to remove the line that was adding the predicate to the (previous local) list.

Reply:
@gillima is right